src/network/rpc_extensions.rs

```diff
@@ -119,6 +120,36 @@ impl LinksByFidRequest {
     }
 }
 
+impl EventsRequest {
+    pub fn timestamps(&self) -> (Option<u64>, Option<u64>) {
+        let start_timestamp = match self.start_timestamp {
```

Comment:
Can you use `clone()`? If so, maybe not worth this level of indirection-- it's easy enough to call clone where needed.